### PR TITLE
Remove auth fallback

### PR DESCRIPTION
**Pull request by enekid**
Remove auth fallback system:
* Generate on the fly api keys if users api keys have not yet been migrated
* Don't use `default_public' api key if provided api key does not exist

closes CartoDB/Windshaft-cartodb#881

### CODE REVIEW

File: test/acceptance/auth/authorization-basic-use-cases.js [hunk 1]
```diff
@@ -122,31 +119,27 @@ describe('authorization fallback', function () {
...
-                keysToDelete['user:user_previous_to_project_auth:mapviews:global'] = 5;
+                keysToDelete['user:localhost:mapviews:global'] = 5;
 
                 done();
             }
         );
     });
 
-    it("succeed with non-existent api key - defaults to default", function (done) {
+    it("fail with non-existent api key", function (done) {
```
Comment (dgaubert) [inline test/acceptance/auth/authorization-basic-use-cases.js]: Do we have a test to check when a request w/o api_key defaults to default public?

File: test/acceptance/auth/authorization-basic-use-cases.js [hunk 2]
```diff
@@ -126,6 +126,22 @@ describe('Basic authorization use cases', function () {
         );
     });
 
+    it("fail with default - sending no api key token", function (done) {
```
Comment (dgaubert) [inline test/acceptance/auth/authorization-basic-use-cases.js]: We need the contrary, we must use `default_public` when no api_key is provided, if not we are going to break a lot of applications, and legit request. We cannot enforce them to use `default_public` w/o any advise or give them a grace period to change their apps or maps.

Comment (enekid) [reply test/acceptance/auth/authorization-basic-use-cases.js]: I've re-organized/rename the tests to reflect that.

**Comment by dgaubert** [comment]
To fix failing test, please synchronize this branch with master.

File: test/acceptance/auth/authorization-basic-use-cases.js (hunk 2 above)
Comment (dgaubert) [reply test/acceptance/auth/authorization-basic-use-cases.js]: I can't see that change.

**Pull request merged**